src/category_theory/limits/shapes/terminal.lean

```diff
@@ -83,13 +84,13 @@ t.hom_ext _ _
 @[simp] lemma is_initial.to_self {X : C} (t : is_initial X) : t.to X = 𝟙 X :=
 t.hom_ext _ _
 
-/-- Any morphism from a terminal object is mono. -/
-lemma is_terminal.mono_from {X Y : C} (t : is_terminal X) (f : X ⟶ Y) : mono f :=
-⟨λ Z g h eq, t.hom_ext _ _⟩
+/-- Any morphism from a terminal object is split mono. -/
+lemma is_terminal.split_mono_from {X Y : C} (t : is_terminal X) (f : X ⟶ Y) : split_mono f :=
```

Comment:
Since these are not instances, I think it's worth it to also keep the original `mono_from` and `epi_to`.

Reply:
Hmm, I'm inclined to disagree - the normal usage (in my experience) of these is that we have `is_terminal` and `haveI` is used to register the new instance, so it's locally used as an instance. Happy to be overruled!

Reply:
I agree, that probably accounts for 90% of the usage (maybe even more). But say you are in a term-mode proof, and you have a goal that is exactly `mono f`, then it would be very convenient if the lemma is right there. On the other hand, maybe `by haveI := t.split_mono_from; apply_instance` is not that bad.
I'm fine either way.

bors d+

Reply:
Good point! I've amended, I'll merge once CI passes